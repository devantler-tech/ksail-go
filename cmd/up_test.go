package cmd_test

import (
	"bytes"
	"testing"

	"github.com/devantler-tech/ksail-go/cmd"
)

func TestNewUpCmd(t *testing.T) {
	// Arrange
	t.Parallel()

	// Act
	cmd := cmd.NewUpCmd()

	// Assert
	if cmd == nil {
		t.Fatal("expected command to be created")
	}

	if cmd.Use != "up" {
		t.Fatalf("expected Use to be 'up', got %q", cmd.Use)
	}

	if cmd.Short != "Start the Kubernetes cluster" {
		t.Fatalf("expected Short description, got %q", cmd.Short)
	}
}

func TestUpCmd_Execute(t *testing.T) {
	// Arrange
	t.Parallel()

	var out bytes.Buffer

	cmd := cmd.NewUpCmd()
	cmd.SetOut(&out)

	// Act
	err := cmd.Execute()

	// Assert
	if err != nil {
		t.Fatalf("expected no error, got %v", err)
	}

	got := out.String()

	expected := "âœ” Cluster created and started successfully (stub implementation)\n"

	if got != expected {
		t.Fatalf("expected output %q, got %q", expected, got)
	}
}
