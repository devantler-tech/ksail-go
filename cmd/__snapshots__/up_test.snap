
[TestUpCmd_Help - 1]
Start the Kubernetes cluster defined in the project configuration.

Usage:
  up [flags]

Flags:
  -c, --context string               Kubernetes context to use (default "kind-ksail-default")
  -d, --distribution Distribution    Kubernetes distribution to use (default Kind)
      --distribution-config string   Configuration file for the distribution (default "kind.yaml")
  -h, --help                         help for up
  -t, --timeout duration             Timeout for cluster operations

---

[TestUpCmd_Execute - 1]
✔ Cluster created and started successfully (stub implementation)
► Distribution: Kind
► Context: kind-ksail-default

---
